File: app/src/main/java/de/schooltec/datapass/datasupplier/CarrierNotSelectedSupplier.java
package de.schooltec.datapass.datasupplier;

import android.content.Context;

/**
 * Supplier which simply says that the carrier is not selected by the user atm.
 *
 * @author Martin Hellwig
 */
class CarrierNotSelectedSupplier extends DataSupplier
{
    @Override
    public ReturnCode getData(Context context)
    {
        return ReturnCode.CARRIER_NOT_SELECTED;
    }

    @Override
    public boolean isRealDataSupplier() {
        return false;
    }

    @Override
    public String getTrafficWasted()
    {
        return "";
    }

    @Override
    public String getTrafficAvailable()
    {
        return "";
    }

    @Override
    public String getTrafficUnit()
    {
        return "";
    }

    @Override
    public int getTrafficWastedPercentage()
    {
        return 0;
    }

    @Override
    public String getLastUpdate()
    {
        return "";
    }
}

File: app/src/main/java/de/schooltec/datapass/datasupplier/CarrierNotSelectedSupplier.kt
package de.schooltec.datapass.datasupplier

import android.content.Context
import java.util.*

/**
 * Supplier which simply says that the carrier is not selected by the user atm.
 *
 * @author Martin Hellwig
 */
internal class CarrierNotSelectedSupplier : DataSupplier {
    override val isRealDataSupplier: Boolean
        get() = false

    override val trafficWasted: Long
        get() = 0

    override val trafficAvailable: Long
        get() = 0

    override val lastUpdate: Date
        get() = Date()

    override fun fetchData(context: Context): DataSupplier.ReturnCode {
        return DataSupplier.ReturnCode.CARRIER_NOT_SELECTED
    }
}
